### PR TITLE
Modifications for Chapel Packaging

### PR DESCRIPTION
**Pull request by ben-albrecht**
Some modifications necessary for packaging Chapel into a dpkg:

* Clean up `*.pyc` files with `make clean/cleanall/clobber`
* Remove executable permission from `runtime/include` headers 
* Add shebang for `chplenv/utils.py`

*Note: These changes could be made as a packaging patch, but keeping the changes upstream (as in the Chapel master branch) seems easier/cleaner*

### CODE REVIEW

File: util/chplenv/utils.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#!/usr/bin/env python
```
Comment (ronawho) [inline util/chplenv/utils.py]: Out of curiosity why is this needed?

Comment (ben-albrecht) [inline util/chplenv/utils.py]: Without it, this file trigger the following lintian error: https://lintian.debian.org/tags/executable-not-elf-or-script.html

Comment (ronawho) [inline util/chplenv/utils.py]: utils.py doesn't have a main function and is only a helper module used by other chplenv scripts. I think we should remove it's execute permission, instead of adding the shebang

Comment (ben-albrecht) [inline util/chplenv/utils.py]: That sounds reasonable. I was just staying consistent. The same could be said for the following files:

* chpl_3p_dlmalloc_configs.py
* chpl_3p_gmp_configs.py
* chpl_3p_hwloc_configs.py
* chpl_3p_massivethreads_configs.py
* chpl_3p_qthreads_configs.py
* chpl_3p_re2_configs.py
* chpl_3p_tcmalloc_configs.py

I'm OK with removing the shebang / executable status from all of these including utils.py.

Comment (ronawho) [inline util/chplenv/utils.py]: I think I'd prefer that

Comment (ben-albrecht) [inline util/chplenv/utils.py]: See latest commit.

**Comment by ronawho** [comment]
looks good to me!
